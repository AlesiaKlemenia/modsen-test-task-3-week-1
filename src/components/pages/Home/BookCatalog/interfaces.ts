export interface ICatalogProps {
  children: JSX.Element[] | JSX.Element;
}
