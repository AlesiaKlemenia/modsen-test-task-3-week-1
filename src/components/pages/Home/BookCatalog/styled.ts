import styled from 'styled-components';

export const Catalog = styled.div`
  width: 97%;
  margin: 1.5rem;

  display: grid;
  align-items: center;
  justify-content: space-around;
  grid-template-columns: repeat(auto-fill, minmax(301px, 350px));
  grid-auto-rows: minmax(430px, 500px);
  grid-gap: 3rem;
`;

export const EmptyCatalog = styled.div`
  width: 0;
  height: 0;
  visibility: none;
`;
