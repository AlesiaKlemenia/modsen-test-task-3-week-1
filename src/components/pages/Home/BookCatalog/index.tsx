import { ICatalogProps } from './interfaces';
import { Catalog } from './styled';

const BooksCatalog = ({ children }: ICatalogProps): JSX.Element => (
  <Catalog>{children}</Catalog>
);

export default BooksCatalog;
