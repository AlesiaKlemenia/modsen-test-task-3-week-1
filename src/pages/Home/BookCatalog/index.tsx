import { ICatalogProps } from '@/pages/Home/BookCatalog/interfaces';
import { Catalog } from '@/pages/Home/BookCatalog/styled';

const BooksCatalog = ({ children }: ICatalogProps): JSX.Element => (
  <Catalog>{children}</Catalog>
);

export default BooksCatalog;
