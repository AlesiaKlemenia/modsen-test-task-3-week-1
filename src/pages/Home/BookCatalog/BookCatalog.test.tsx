import '@testing-library/jest-dom';

import { StyledEngineProvider } from '@mui/material';
import renderer from 'react-test-renderer';

import BookCatalog from '@/pages/Home/BookCatalog';

describe('Book Catalog component', () => {
  it('renders correctly with children', () => {
    const componentRenderer = renderer.create(
      <StyledEngineProvider injectFirst>
        <BookCatalog>
          <div className="test" />
        </BookCatalog>
      </StyledEngineProvider>,
    );
    const tree = componentRenderer.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
