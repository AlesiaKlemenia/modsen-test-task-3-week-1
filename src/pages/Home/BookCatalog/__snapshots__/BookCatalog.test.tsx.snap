// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Catalog component renders correctly with children 1`] = `
<div
  className="sc-beySPh eQAjHc"
>
  <div
    className="test"
  />
</div>
`;
